src/main/java/org/traccar/protocol/NavtelecomProtocolDecoder.java

```diff
@@ -208,6 +208,19 @@ protected Object decode(
                                 case 3:
                                     position.setDeviceTime(new Date(buf.readUnsignedIntLE() * 1000));
                                     break;
+                                case 4:
+                                    value = buf.readUnsignedByte();
+                                    position.set(Position.KEY_ALARM, BitUtil.check(value, 2));
+                                    int guardMode = BitUtil.between(value, 3, 4);
+                                    position.set(
+                                            Position.KEY_ARMED,
+                                            (0 < guardMode) && (guardMode < 3));
```

Comment:
The logic here seems overcomplicated for just checking 2 bits.

Reply:
We just have 2 bits allocated for security modes and the description is as follows:
0 - no protection
1 - security
2 - additional security
3 - service

What's the best way to remake?

Reply:
OK, if that's how it works, I guess it makes sense to keep as is. Does it not fit on a single line?

Reply:
_Does it not fit on a single line?_
No, it fits, I don’t know why I did it =) I’ll correct it.